Drop-In/src/androidTest/java/com/braintreepayments/api/CardDetailsFragmentUITest.kt

```diff
@@ -0,0 +1,380 @@
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(withId(R.id.bt_card_form_expiration)).check(matches(isFocused()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_displaysCVVAndPostalCodeWhenPresentInConfiguration() {
+        val args = Bundle()
+        args.putParcelable("EXTRA_DROP_IN_REQUEST", DropInRequest().clientToken(Fixtures.CLIENT_TOKEN))
+        args.putParcelable("EXTRA_CARD_FORM_CONFIGURATION", CardFormConfiguration(true, true))
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(isRoot()).perform(waitFor(500))
+
+        onView(withId(R.id.bt_card_form_cvv)).check(matches(isDisplayed()))
+        onView(withId(R.id.bt_card_form_postal_code)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_hidesCVVandPostalCodeWhenNotPresentInConfiguration() {
+        val args = Bundle()
+        args.putParcelable("EXTRA_DROP_IN_REQUEST", DropInRequest().clientToken(Fixtures.CLIENT_TOKEN))
+        args.putParcelable("EXTRA_CARD_FORM_CONFIGURATION", CardFormConfiguration(false, false))
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(isRoot()).perform(waitFor(500))
+
+        onView(withId(R.id.bt_card_form_cvv)).check(matches(not(isDisplayed())))
+        onView(withId(R.id.bt_card_form_postal_code)).check(matches(not(isDisplayed())))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_withDropinRequestRequiringCardholderName_setsCardFormsCardholderNameToRequired() {
+        val args = Bundle()
+        args.putParcelable("EXTRA_DROP_IN_REQUEST", DropInRequest().clientToken(Fixtures.CLIENT_TOKEN).cardholderNameStatus(FIELD_REQUIRED))
+        args.putParcelable("EXTRA_CARD_FORM_CONFIGURATION", CardFormConfiguration(false, false))
+        args.putString("EXTRA_CARD_NUMBER", VISA)
+        val scenario = FragmentScenario.launchInContainer(CardDetailsFragment::class.java, args, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(isRoot()).perform(waitFor(500))
+
+        onView(withId(R.id.bt_card_form_expiration)).perform(typeText(ExpirationDate.VALID_EXPIRATION))
+        scenario.onFragment { fragment ->
+            assertFalse(fragment.cardForm.isValid)
+        }
+
+        onView(withId(R.id.bt_card_form_cardholder_name)).perform(typeText("Brian Tree"))
+        scenario.onFragment { fragment ->
+            assertTrue(fragment.cardForm.isValid)
```

Comment:
Would this test be more powerful in checking that the Cardholder name is require by not typing in "Brian Tree" and then asserting the CardForm is Not valid?

Reply:
Yeah we do have an assertion to assert that card form `isValid = false` in the block above.

Reply:
Ah - I missed that, thanks!